File: tms-safety-api/src/main/java/com/svtrucking/logistics/dto/DeviceRegisterDto.java
package com.svtrucking.logistics.dto;

import com.svtrucking.logistics.enums.DeviceStatus;
import java.time.LocalDateTime;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class DeviceRegisterDto {

  private Long id;
  private Long driverId;
  private String driverName;
  private String deviceId;
  private String deviceName;
  private String os;
  private String version;
  private String appVersion;
  private String manufacturer;
  private String model;
  private String ipAddress;
  private String location;
  private DeviceStatus status;
  private LocalDateTime registeredAt;
  private String approvedBy;
  private LocalDateTime statusUpdatedAt;
}

File: tms-admin-web-ui/src/app/models/device-register.dto.ts
export interface DeviceRegisterDto {
  id: number;
  driverId: number | null;
  driverName: string;

  deviceId: string;
  deviceName: string;
  os: string;
  version: string;
  appVersion: string;
  manufacturer: string;
  model: string;
  ipAddress: string;
  location: string;

  status: 'PENDING' | 'APPROVED' | 'BLOCKED'; // Match Java enum DeviceStatus

  registeredAt: string; // ISO date-time string (LocalDateTime from backend)
  approvedBy: string;
  statusUpdatedAt: string; // ISO date-time string (LocalDateTime from backend)

  showMenu?: boolean; // Local UI-only field
}
